sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystConf.scala

```diff
@@ -51,6 +52,7 @@ case class SimpleCatalystConf(
     caseSensitiveAnalysis: Boolean,
     orderByOrdinal: Boolean = true,
     groupByOrdinal: Boolean = true,
+    optimizerMetadataOnly: Boolean = true,
```

Comment:
that's a good point. The new rule is in sql core module and can access the `SQLConf`, we don't need to add the conf in `CatalystConf`